templates/sync-function/expiry-module.js

```diff
@@ -0,0 +1,35 @@
+function expirationModule(utils) {
+  return {
+    setDocExpiry: function(doc, oldDoc, expiryValue) {
+      var resolvedExpiryValue = utils.resolveDocumentConstraint(expiryValue);
+
+      if (typeof resolvedExpiryValue === 'number') {
+        // The positive sign is applied to numeric values as a workaround to this Sync Gateway issue:
+        // https://github.com/couchbase/sync_gateway/issues/3452
+        expiry(+resolvedExpiryValue);
```

Comment:
This [bug](https://github.com/couchbase/sync_gateway/issues/3452) really threw me for a loop when I was trying to manually verify the behaviour of the `expiry` constraint, but at least the workaround is relatively non-invasive.